plugins/statuslines/lualine.nix

```diff
@@ -0,0 +1,129 @@
+{ pkgs, config, lib, ... }:
+with lib;
+let
+  cfg = config.programs.nixvim.plugins.lualine;
+  helpers = import ../helpers.nix { lib = lib; };
+  separators = mkOption {
+    type = types.submodule {
+      options = {
+        left = mkOption {
+          default = " ";
+          type = types.str;
+          description = "left separator";
+        };
+        right = mkOption {
+          default = " ";
+          type = types.str;
+          description = "right separator";
+        };
+      };
+    };
+    default = { };
+  };
+  component_options = mode:
+    mkOption {
+      type = types.submodule {
+        options = {
+          mode = mkOption {
+            type = types.str;
+            default = "${mode}";
+          };
+          icons_enabled = mkOption {
+            type = types.enum [ "True" "False" ];
```

Comment:
This should be a boolean! Is there any reason in particular you did it this way?

Reply:
Somehow, when I make it a boolean neovim says it can't resolve this component.

Reply:
Interesting, I'll try it out